batch/batch/driver/instance_collection/pool.py

```diff
@@ -227,15 +270,65 @@ async def create_instances_from_ready_cores(self, ready_cores_mcpu, region=None)
                         data_disk_size_gb=self.data_disk_size_gb,
                         region=region,
                     )
-                    for _ in range(instances_needed)
+                    for _ in range(n_instances)
                 ]
             )
 
-    async def create_instances(self):
-        if self.app['frozen']:
-            log.info(f'not creating instances for {self}; batch is frozen')
-            return
+    async def create_instances_from_ready_cores(self, ready_cores_mcpu, region=None):
+        instances_needed = self.compute_n_instances_needed(
+            ready_cores_mcpu,
+            self.n_instances_by_state,
+            region,
+        )
+        await self._create_instances(instances_needed, region)
+        return instances_needed
+
+    async def ready_cores_mcpu_from_estimated_job_queue(self):
+        n_free_cores = int(
+            10 * 8 * self.worker_cores
```

Comment:
Let's make `10` (and its usage above in `compute_n_instances_needed`) a named constant

Reply:
Also, I either don't understand what we're computing here or I don't think it's correct. If we're trying to compute the upper bound on how many cores we would have available to schedule on in a few minutes, then we can write:

```python
autoscaler_runs_per_minute = 60 / AUTOSCALER_LOOP_PERIOD_SECONDS
max_new_cores_in_two_and_a_half_minutes = int(2.5 * MAX_INSTANCES_PER_AUTOSCALER_LOOP * autoscaler_runs_per_minute)

max_possible_future_available_cores = self.worker_cores + max_new_cores_in_two_and_a_half_minutes
```

I would rather use a name like `max_possible_future_available_cores` than `n_free_cores`